src/ex1/script.js

```diff
@@ -0,0 +1,129 @@
+const addTaskButton = document.querySelector(".add-task");
+const taskInput = document.querySelector("#task-text");
+const allTasksContainer = document.querySelector(".all-tasks-container");
+const pendingTasksText = document.querySelector(".pending-tasks");
+let pendingTasks = 0;
+const clearButton = document.querySelector(".clear-all");
+const sortButton = document.querySelector(".name-sort");
+
+// Event trigger when user clicks the add task button
+addTaskButton.addEventListener("click", () => {
+   addTask();
+});
+
+// Same as the button add task but only when user press "Enter"
+taskInput.addEventListener("keypress", (e) => {
+   if (e.key === "Enter") {
+      addTask();
+   }
+});
+
+// Clears all tasks when clear all button is clicked
+clearButton.onclick = () => {
+   document
+      .querySelectorAll(".task-container")
+      .forEach((task) => task.remove());
+   pendingTasksUpdate(0);
+};
+
+// Sorts the tasks containers by name
+sortButton.onclick = () => {
+   const taskContainers = [...document.querySelectorAll(".task-container")];
+   taskContainers.sort((a, b) =>
+      a.children[0].innerText.localeCompare(b.children[0].innerText)
+   );
+   taskContainers.map((container) => container.remove());
+   taskContainers.map((container) => allTasksContainer.appendChild(container));
+};
+
+// Adds the user task input to the tasks list
+function addTask() {
+   // Checks for blank spaces
+   if (taskInput.value.trim().length === 0) {
+      alert("Please fill in a task");
+      taskInput.value = "";
```

Comment:
redundant input clear (the input is empty)

Reply:
I agree, it is empty but in case of a few blank spaces I prefer to delete them. If there are a few blank spaces the placeholder message isn't shown as well.